llama-index-core/tests/readers/test_json.py

```diff
@@ -70,3 +70,26 @@ def test_jsonl() -> None:
         assert data[1].get_content().index("test2") is not None
         assert isinstance(data[2].get_content(), str)
         assert data[2].get_content().index("test3") is not None
+
+
+def test_clean_json() -> None:
+    """Test JSON reader using the clean_json function."""
+    with TemporaryDirectory() as tmp_dir:
+        file_name = f"{tmp_dir}/test5.json"
+        with open(file_name, "w") as f:
+            f.write('{ "a": { "b": "c" } }')
+
+        # If levels back is set clean_json is ignored
+        reader1 = JSONReader(levels_back=0, clean_json=False)
+        data1 = reader1.load_data(file_name)
+        assert data1[0].get_content() == "a b c"
+
+        # If clean_json is false the full json should be contained in a document
+        reader1 = JSONReader(clean_json=False)
+        data1 = reader1.load_data(file_name)
+        assert data1[0].get_content() == '{"a": {"b": "c"}}'
+
+        # If clean_json is True the full json should be contained in a document
+        reader1 = JSONReader(clean_json=True)
+        data1 = reader1.load_data(file_name)
+        assert data1[0].get_content() == '"a": {\n"b": "c"'
```

Comment:
is this the expected output for clean_json?

Reply:
It is and that's the existing behavior. A product of the regex ` useful_lines = [ line for line in lines if not re.match(r"^[{}\[\],]*$", line) ] `. This PR is really to add the option to not clean which returns the full valid JSON object

Reply:
should we just remove this broken json output. since clean_json is kinda of confusing. I suppose to get right json output when clean_json = true

Reply:
I agree it is a bit confusing, and would welcome suggestions. I see the value of both options, full JSON object and a minimized output has reduced characters of little meaning when creating embeddings.  I'm guessing the regex is setup to only remove characters like braces when there is no other text on the same line as to not remove characters included in a string on accident. Ex) the initial JSON was `{"a":"b{35"}`.

Reply:
thanks @thegunslingers I think we can merge this for now. But this part needs more polihs